nixos/modules/services/security/nwaku.nix

```diff
@@ -0,0 +1,278 @@
+            description = mdDoc "Listening port of the REST HTTP server.";
+          };
+        };
+
+        rpc = {
+          enable = lib.mkEnableOption (mdDoc "Enable Waku JSON-RPC server.");
+          admin = mkOption {
+            type = types.bool;
+            default = false;
+            description = mdDoc "Enable access to JSON-RPC Admin API.";
+          };
+          private = mkOption {
+            type = types.bool;
+            default = false;
+            description = mdDoc "Enable access to JSON-RPC Private API.";
+          };
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Listening address of the JSON-RPC server.";
+          };
+          port = mkOption {
+            type = types.int;
+            default = 5052;
+            description = mdDoc "Listening port of the JSON-RPC server.";
+          };
+        };
+
+        metrics = {
+          enable = lib.mkEnableOption (mdDoc "Nim Waku node metrics endpoint");
+          logging = mkOption {
+            type = types.bool;
+            default = false;
+            description = mdDoc "Enable metrics logging.";
+          };
+          address = mkOption {
+            type = types.str;
+            default = "127.0.0.1";
+            description = mdDoc "Listening address of the metrics server.";
+          };
+          port = mkOption {
+            type = types.int;
+            default = 8008;
+            description = mdDoc "Listening HTTP port of the metrics server.";
+          };
+        };
+
+        extraArgs = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          example = ["--eth-client-address=ws://localhost:8540/"];
+          description = mdDoc "Additional arguments passed to node.";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = cfg.nodekey != null -> stringLength cfg.nodekey == 64;
+        message = "Nim Waku node key needs to be 64 characters long!"; }
+      { assertion = cfg.dnsDiscovery.enable -> cfg.dnsDiscovery.url != null;
+        message = "Nim Waku node requires ENR URL for DNS Discovery!"; }
+      { assertion = cfg.v5Discovery.enable -> cfg.v5Discovery.bootstrapNode != null;
+        message = "Nim Waku node requires a bootstrap node for V5 Discovery!"; }
+    ];
+
+    systemd.services.nwaku = {
+      enable = true;
+      description = "Nim Waku node";
+
+      serviceConfig = {
+        DynamicUser = true;
+        # Hardening measures
+        PrivateTmp = "true";
+        ProtectSystem = "full";
+        NoNewPrivileges = "true";
+        PrivateDevices = "true";
+        MemoryDenyWriteExecute = "true";
+
+        StateDirectory = "nwaku";
+        RuntimeDirectory = "nwaku";
+        WorkingDirectory = cfg.dbPath;
+        Restart = "always";
+
+        ExecStart = concatStringsSep " \\\n  " ([
+          "${cfg.package}/bin/wakunode2"
```

Comment:
Given `makeFlags` in `nwaku/default.nix` accepts one of `wakunode1`, `wakunode2`, `wakubridge`, is it okay to hardcode `wakunode2` here?

Reply:
I think so, since the flags are specific to just that service, and it's the only one that people might want to run. The package has those as options purely because it was simple to do that.